cas-server-documentation/installation/Configuring-SAML2-Authentication.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: default
+title: CAS - SAML2 Authentication
+---
+
+# SAML2 Authentication
+CAS can act as a SAML2 identity provider accepting authentication requests and producing SAML assertions.
+
+## Overview
+
+### SAML Endpoints
+The following CAS endpoints respond to supported SAML2 profiles:
+
+- `/cas/idp/profile/SAML2/Redirect/SSO`
+- `/cas/idp/profile/SAML2/POST/SSO`
+
+### IdP Metadata
+The following CAS endpoints handle the generation of SAML2 metadata:
+ 
+- `/cas/idp/metadata`
+
+This endpoint will display the CAS IdP SAML2 metadata upon receiving a GET request. If metadata is already available and generated,
+it will be displayed. If metadata is absent, one will be generated automatically. 
+CAS configuration below dictates where metadata files/keys will be generated and stored.
+
+<div class="alert alert-info"><strong>Review Metadata</strong><p>Due to the way CAS handles the generation of metadata via external 
+libraries, the generated metadata MUST be reviewed and massaged slightly to match the CAS configuration. All other elements MUST be 
```

Comment:
Ah, great question. 

You see, when we generate the metadata from that endpoint, the result that we get has a whole bunch of other elements/bindings in the metadata that we do not want, because we do not support them. Now, since we don't support them, they don't do you any harm but because they are in your metadata and advertised, other SPs mighr think the CAS IDP actually supports them (because that is the point of metada amongst other things), so the confusion. Also, the generated metadata assumes a lot of other things about ports, host names, etc which may not be appropriate for CAS deployment. 

So, why don't we generate the "right" metadata in the first? Because I have no control over it :) 
See https://issues.shibboleth.net/jira/browse/IDP-894

So what this all means is, once you generate the metadata you need to review it to make sure it's correctly defined and matches the configuration. Thus the example provided in the docs for you to compare. You should always review the metadata generated anyway; that's a common practice.